### PR TITLE
[KBK-1] aka ACADEMIC-16455: user role in feedback

### PR DESCRIPTION
**Pull request by aethant**
**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [x] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

### CODE REVIEW

File: ai_aside/constants.py [hunk 1]
```diff
@@ -7,3 +7,32 @@
...
+# The country code determined from the user's request IP address.
+ATTR_KEY_REQUEST_COUNTRY_CODE = 'edx-platform.request_country_code'
+# Whether the user is authenticated or anonymous.
+ATTR_KEY_IS_AUTHENTICATED = 'edx-platform.is_authenticated'
+# The personally identifiable user ID.
+ATTR_KEY_USER_ID = 'edx-platform.user_id'
+# The username.
+ATTR_KEY_USERNAME = 'edx-platform.username'
+# Whether the user is enrolled in the course as a Beta Tester.
+ATTR_KEY_USER_IS_BETA_TESTER = 'edx-platform.user_is_beta_tester'
```
Comment (ashultz0) [inline ai_aside/constants.py:30]: not used

**Comment by ashultz0** [comment]
this needs an update in the CHANGELOG and to try this out we need to make a new version which is here https://github.com/edx/ai-aside/blob/main/ai_aside/__init__.py - this would be a 3.6 because it's not a breaking change but it is New Stuff, and that will go automatically to stage (version 3.x.x) once we cut the version but not to production (version 3.5.x)

**Review by ashultz0 — APPROVED** [review]
I don't think the individual diffs are going to be very useful to future readers so before merge please collapse them together, but good to go.

File: ai_aside/block.py [hunk 2]
```diff
@@ -133,15 +135,19 @@ def _check_summarizable(block):
...
 
-def _render_hook_fragment(handler_url, block, summary_items):
+def _render_hook_fragment(user_role_string, handler_url, block, summary_items):
     """
     Create hook Fragment from block and summarized children.
 
     Gathers data for the summary hook HTML, passes it into _render_summary
     to get the HTML and packages that into a Fragment.
     """
+    print(dir(block))
```
Comment (schenedx) [inline ai_aside/block.py:145]: Please remove this debug line

File: ai_aside/block.py [hunk 3]
```diff
@@ -274,6 +285,28 @@ def _summary_handler_url(self):
...
+        Determine and construct the user_role string that gets injected into the block.
+        """
+        user_role = 'unknown'
+        user = user_service.get_current_user()
+        if user is not None:
+            user_role = user.opt_attrs.get(ATTR_KEY_USER_ROLE)
+            user_enrollment = credit_service.get_credit_state(
+                user.opt_attrs.get(ATTR_KEY_USER_ID), course_key)
+
+            if user_enrollment.get('enrollment_mode') is not None:
```
Comment (schenedx) [inline ai_aside/block.py:306]: #NIT I recognize that if this condition is false, the return of the function is still `unknown`. But the logic on this point is pretty difficult to read. 
Maybe use an `else` statement to make it easier to read?

File: tests/test_block.py [hunk 4]
```diff
@@ -303,20 +307,48 @@ def test_render_hook_fragment(self):
...
+        returned_role = SummaryHookAside._user_role_string_from_services(user_service, credit_service, "course_key")
+        self.assertEqual(returned_role, expected_role)
+
+        self.assertEqual(credit_service.mock_calls,
+                         [call.get_credit_state('the_user_role', 'course_key'),
+                          call.get_credit_state().get('enrollment_mode'),
+                          call.get_credit_state().get('enrollment_mode')])
+
+        user_service.get_current_user.return_value = None
+        expected_role = 'unknown'
```
Comment (schenedx) [inline tests/test_block.py:347]: This should be a new test case or new test function.
You don't need to get into this now, but a way to put multiple test cases into a single test function is using [ddt](https://ddt.readthedocs.io/en/latest/) or other python library.

**Review by schenedx — APPROVED** [review]
I don't have objections to merge. I do have feedback

**Pull request merged**